### PR TITLE
feat: mailing system added

### LINKED ISSUE #72: feat: automatic mailing system
I can make an automatic mailing system that will send a mail to a new user after signup.. 

Can you assign this issue to me with the gssoc tag?

### PR DESCRIPTION
**Pull request by SINJINI-G**
fixes #72 

- mailing service added
- signup updated

@PiyushKalyanpy

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
   },
   "dependencies": {
     "@reduxjs/toolkit": "^1.9.3",
+    "@types/nodemailer": "^6.4.8",
```
Comment (lalitmee) [inline package.json:13]: I think `types` should be installed as `devDependencies`

File: service/mail.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import nodemailer from "nodemailer";
+
+const SOURCE_EMAIL = process.env.SOURCE_EMAIL;
+const SOURCE_PASSWORD = process.env.SOURCE_EMAIL_PASSWORD;
+
+const Transporter = nodemailer.createTransport({service:"gmail", auth:{user:SOURCE_EMAIL, pass:SOURCE_PASSWORD}})
+
+const sendWelcomeMail = async (email:string) => {
+    console.log(email, SOURCE_EMAIL);
```
Comment (lalitmee) [inline service/mail.ts:9]: we should remove this console?

**Review by lalitmee — CHANGES_REQUESTED** [review]
Can you please share some screenshots of video of this working?

**Review by PiyushKalyanpy — CHANGES_REQUESTED** [review]

File: service/mail.ts (hunk 2 above)
Comment (SINJINI-G) [reply service/mail.ts:9]: yes sure!

File: package-lock.json
[the remark this answers is not in the document]
Comment (SINJINI-G) [reply package-lock.json:1]: Deleted the file. Kindly check.

**Comment by SINJINI-G** [comment]
hcoonlbom@logicstreak.com

File: package.json (hunk 1 above)
Comment (SINJINI-G) [reply package.json]: Done

**Comment by SINJINI-G** [comment]
> Can you please share some screenshots of video of this working?

## As you can see the mail is being sent..(in the last frame) whenever a user signs up. Check the Terminal.

poc

**Comment by SINJINI-G** [comment]
Hi @PiyushKalyanpy @lalitmee  ! Can you please review my pr? I had resolved all the conflicts but since it is not merged for a long time, new conflicts are arising.. moreover the gssoc leader board will be opening soon.. It will be really helpful if you could merge the pr and attach the gssoc and label tags

File: lib/mail.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import nodemailer from "nodemailer";
+
+const SOURCE_EMAIL = "";
+const SOURCE_PASSWORD = "";
+
+const Transporter = nodemailer.createTransport({service:"gmail", auth:{user:SOURCE_EMAIL, pass:SOURCE_PASSWORD}})
+
+const sendWelcomeMail = async (email:string) => {
+    
+    console.log(email, SOURCE_EMAIL);
```
Comment (PiyushKalyanpy) [inline lib/mail.ts:10]: remove all the console statements from all files

File: pages/api/mailer.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import type { NextApiRequest, NextApiResponse } from 'next';
+import sendWelcomeMail from '@/lib/mail';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+    if (req.method === 'POST') {
+        console.log(req.body);
```
Comment (PiyushKalyanpy) [inline pages/api/mailer.ts:7]: Remove this also.

File: service/mail.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+      html: `<b>Hi!!</b>,<br/><br/>
+      Congratulations! Your registration for <b>GyanaGuru</b> is successful.<br/><br/>
+      Regards<br/>
+      GyanaGuru<br/>`
+    });
+    console.log(info);
+}
+
+export default sendWelcomeMail
```
Comment (PiyushKalyanpy) [inline service/mail.ts:23]: Add a semicolon at last.

File: package-lock.json
[the remark this answers is not in the document]
Comment (PiyushKalyanpy) [reply package-lock.json:1]: Deleting a file is not a solution, You have to remove it from your commit. This is what your pull request shows. Do you think that it is a good pull request??
image

File: lib/mail.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import nodemailer from "nodemailer";
+
+const SOURCE_EMAIL = "";
+const SOURCE_PASSWORD = "";
```
Comment (PiyushKalyanpy) [inline lib/mail.ts:4]: instead of using string as literal use env variables - 

- process.env.NEXT_SOURCE_EMAIL
- process.env.NEXT_SOURCE_PASSWORD

**Review by PiyushKalyanpy — CHANGES_REQUESTED** [review]

File: pages/signup.tsx [hunk 7]
```diff
@@ -68,13 +69,13 @@ const SignUp = () => {
     } else {
       signup(userData.email, userData.password).then(({ data }: any) => {
         console.log("aa",data);
+        await sendEmail(user.email)
```
Comment (PiyushKalyanpy) [inline pages/signup.tsx:72]: Await functions are only allowed in async functions. Please define async.

**Comment by PiyushKalyanpy** [comment]
Hey there, 
We are closing this pull request as the changes we requested were not implemented. Please create a new PR once you implement the changes.

**Pull request closed without merging**